src/components/Common/SearchByMultipleFields.tsx

```diff
@@ -0,0 +1,259 @@
+  useMemo,
+} from "react";
+import { useTranslation } from "react-i18next";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { cn } from "@/lib/utils";
+import CareIcon from "@/CAREUI/icons/CareIcon";
+import PhoneNumberFormField from "@/components/Form/FormFields/PhoneNumberFormField";
+import { FieldError } from "@/components/Form/FieldValidators";
+import {
+  Command,
+  CommandGroup,
+  CommandItem,
+  CommandList,
+} from "@/components/ui/command";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+interface SearchOption {
+  key: string;
+  label: string;
+  type: "text" | "phone";
+  placeholder: string;
+  value: string;
+  shortcutKey: string;
+  component?: React.ComponentType<HTMLDivElement>;
+}
+
+interface SearchByMultipleFieldsProps {
+  options: SearchOption[];
+  onSearch: (key: string, value: string) => void;
+  initialOption?: SearchOption;
+  className?: string;
+  inputClassName?: string;
+  buttonClassName?: string;
+}
+
+type EventType = {
+  value: string;
+  target?: { value: string };
+};
+
+const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
+  options,
+  onSearch,
+  initialOption,
+  className,
+  inputClassName,
+  buttonClassName,
+}) => {
+  const { t } = useTranslation();
+  const [selectedOption, setSelectedOption] = useState<SearchOption>(
+    initialOption || options[0],
+  );
+  const [searchValue, setSearchValue] = useState(selectedOption.value || "");
+  const [open, setOpen] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [focusedIndex, setFocusedIndex] = useState(0);
+  const [error, setError] = useState<string | undefined | boolean>();
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "/" && document.activeElement !== inputRef.current) {
+        e.preventDefault();
+        setOpen(true);
+      } else if (e.key === "ArrowDown") {
+        setFocusedIndex((prevIndex) =>
+          prevIndex === options.length - 1 ? 0 : prevIndex + 1,
+        );
+      } else if (e.key === "ArrowUp") {
+        setFocusedIndex((prevIndex) =>
+          prevIndex === 0 ? options.length - 1 : prevIndex - 1,
+        );
+      } else if (e.key === "Enter") {
+        handleOptionChange(options[focusedIndex]);
+      }
+
+      if (e.key === "Escape") {
+        inputRef.current?.focus();
+        setOpen(false);
+      }
+
+      options.forEach((option) => {
+        if (e.key === "/") {
+          setOpen(true);
+        } else if (
+          e.key.toLocaleLowerCase() ===
+            option.shortcutKey.toLocaleLowerCase() &&
+          open
+        ) {
+          e.preventDefault();
+          handleOptionChange(option);
+        }
+      });
+    };
+
+    document.addEventListener("keydown", handleKeyDown);
+    return () => document.removeEventListener("keydown", handleKeyDown);
+  }, [focusedIndex, open]);
```

Comment:
Add all dependencies for the react hook
```suggestion
}, [focusedIndex, open, handleOptionChange, options]);
```